src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -91,10 +84,19 @@ private void ParseCacheContent(JObject contentJobject)
             if (contentJobject.TryGetValue(nameof(Version), StringComparison.OrdinalIgnoreCase, out JToken versionToken))
             {
                 Version = versionToken.ToString();
+
+                if (Version != CurrentVersion)
+                {
+                    _environmentSettings.Host.LogDiagnosticMessage(
+                        $"Template cache file version is {Version}, but template engine is {CurrentVersion}, rebuilding cache.",
+                        "Debug");
```

Comment:
We are not doing anything to rebuild the cache here, but the message says we are. If someone calls this method from somewhere else with no intention of rebuilding the cache in case of a version mismatch, this message will be incorrectly shown. Should we change the text or move this log line out?

Reply:
I agree here